### ISSUE TITLE
`import librosa` slow to run at 48 seconds

### ISSUE THREAD

**Issue by fabswt** [librosa/librosa#1541]
**Describe the bug**

`import librosa` takes way too long, at 48 seconds.

**To Reproduce**

```python
import librosa

# Start a stopwatch to time it.
# Stop the stopwatch when the Python prompt becomes responsive/available again.
# In my case, it takes 48 seconds.
```

**Expected behavior**
Would expect it to go fast.

**Software versions**

```txt
Python 3.8.10 (default, Mar 15 2022, 12:22:08)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.

>>> import platform; print(platform.platform())
Linux-5.10.104-linuxkit-x86_64-with-glibc2.29

>>> import sys; print("Python", sys.version)
Python 3.8.10 (default, Mar 15 2022, 12:22:08)
[GCC 9.4.0]

>>> import numpy; print("NumPy", numpy.__version__)
NumPy 1.22.4

>>> import scipy; print("SciPy", scipy.__version__)
SciPy 1.8.1

>>> import librosa; print("librosa", librosa.__version__)
librosa 0.9.2

>>> librosa.show_versions()
INSTALLED VERSIONS
------------------
python: 3.8.10 (default, Mar 15 2022, 12:22:08)
[GCC 9.4.0]

librosa: 0.9.2

audioread: 2.1.9
numpy: 1.22.4
scipy: 1.8.1
sklearn: 1.1.1
joblib: 1.1.0
decorator: 5.1.1
soundfile: 0.10.3
resampy: 0.3.1
numba: 0.55.2
pooch: v1.6.0
packaging: 21.3

numpydoc: None
sphinx: None
sphinx_rtd_theme: None
sphinx_multiversion: None
sphinx_gallery: None
mir_eval: None
ipython: None
sphinxcontrib-svg2pdfconverter: None
pytest: None
pytest-mpl: None
pytest-cov: None
matplotlib: None
samplerate: None
soxr: None
contextlib2: None
presets: None
```

System:

- I'm running things inside of a Docker container running `heroku/heroku:20` (based off Ubuntu 20).
  - If I run `import librosa` from the host computer running macOS the speed is normal.
  - The Mac is a 2021 MacBook Pro M1.

**Additional context**

What I've tried:

- Restarting my Docker container;
- Uninstalling / Reinstalling librosa

There's no error message in the prompt. Where could I look for additional information? (Say logs, or how to run libsndfile or other from the command-line?)

**Comment by lostanlen** [comment]
Hello @fabswt, does this wait time happen on every `import` or only the first import after installation?

**Comment by fabswt** [comment]
Hey @lostanlen,

Two usages:

- In the Python prompt:
  - The slow load time happens any time I `import librosa` in a new Python prompt (but not if I reimport in the same prompt.) So this happens not just after install, but in any new Python prompt.
- In the Flask app where I'm trying to use it:
  - The slow load time happens any time I reload the webapp after saving the Python code, even if I actually didn't change anything to the code (but not, say, after editing a Jinja template.)

**Comment by bmcfee** [comment]
I'm pretty sure this is coming from the 0.3 release of resampy, which has much more advanced numba compilation than the 0.2 series.  At some point in the next few days, I'll try to put out a patch that caches the jit compilation, so this becomes a one-time cost rather than an every-import cost.

#1523 will also go quite a ways toward speeding up imports in the future.

FWIW, here's the current dev branch's import time log:
```
$ python -X importtime -c "import librosa" 2> librosa-import.log
$ tuna librosa-import.log
```
![image](https://user-images.githubusercontent.com/1190540/180224754-bc6b2bc1-9f3e-43fb-a997-e582ada6246a.png)

**Comment by bmcfee** [comment]
Ok, thanks for confirmation.

For a bit more context, we ran into some pretty nasty segfaults when using jit caches in resampy previously, which is why this was disabled in the 0.3 release.  I'll try to dig into it more to see if we can sort it out.

> * Not great, but better.

This I think can't be avoided until we implement lazy-import in the 0.10 series.  I'd like to wait until the lazy-import module has a proper release (it's currently at 0.1rc2), but since it's a dependency for scipy 1.9, I expect that to happen pretty soon.

**Comment by bmcfee** [comment]
This will be resolved with the 0.4 release of resampy next week, and requires no further action from the librosa side.

**Comment by bmcfee** [comment]
@fabswt Just a quick heads up that the 0.4 release of resampy shipped on monday, and that should fix this particular import issue.
